src/VirtualClient/VirtualClient.Actions/PostgreSQL/PostgreSQLServerExecutor.cs

```diff
@@ -354,6 +368,210 @@ private async Task CreateDatabaseAsync(EventContext telemetryContext, Cancellati
+                        "of the existing disks.",
+                        ErrorReason.DependencyNotFound);
+                }
+
+                if (await this.CreateMountPointsAsync(disksToTest, cancellationToken).ConfigureAwait(false))
+                {
+                    // Refresh the disks to pickup the mount point changes.
+                    await Task.Delay(1000).ConfigureAwait(false);
+                    IEnumerable<Disk> updatedDisks = await this.SystemManagement.DiskManager.GetDisksAsync(cancellationToken)
+                        .ConfigureAwait(false);
+
+                    disksToTest = DiskFilters.FilterDisks(updatedDisks, diskFilter, this.Platform).ToList();
+                }
+
+                foreach (Disk disk in disksToTest)
+                {
+                    if (disk.GetPreferredAccessPath(this.Platform) != "/mnt")
+                    {
+                        diskPathsArgument += $"{disk.GetPreferredAccessPath(this.Platform)} ";
+                    }
+                }
+
+                if (this.Platform == PlatformID.Unix)
+                {
+                    string balancedScript = "balanced.sh";
+                    string workingDirectory = null;
+
+                    LinuxDistributionInfo distroInfo = await this.SystemManagement.GetLinuxDistributionAsync(cancellationToken);
+
+                    switch (distroInfo.LinuxDistribution)
+                    {
+                        case LinuxDistribution.Ubuntu:
+                        case LinuxDistribution.Debian:
+                            workingDirectory = this.Combine(this.PostgreSqlPackagePath, "ubuntu");
+                            break;
+                    }
+
+                    using (IProcessProxy process = await this.ExecuteCommandAsync(
+                        this.Combine(workingDirectory, balancedScript),
+                        diskPathsArgument,
+                        workingDirectory,
+                        telemetryContext,
+                        cancellationToken,
+                        runElevated: true))
+                    {
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            await this.LogProcessDetailsAsync(process, telemetryContext, "PostgreSQL", logToFile: true);
+                            process.ThrowIfWorkloadFailed();
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates mount points for any disks that do not have them already.
+        /// </summary>
+        /// <param name="disks">This disks on which to create the mount points.</param>
+        /// <param name="cancellationToken">A token that can be used to cancel the operation.</param>
+        private async Task<bool> CreateMountPointsAsync(IEnumerable<Disk> disks, CancellationToken cancellationToken)
```

Comment:
Do we need to do this? Does FormatDisk work? Could you describe the scenario where you need this?